webpages/js/metaanalyses.js

```diff
@@ -626,69 +673,116 @@
+      forestGroupT.parentElement.insertBefore(forestGroupEl, forestGroupT);
 
-      currY += parseInt(plotEl.dataset.lineHeight);
-    })
+      // put experiments into the plot
+      lines.forEach(function (line) {
+        if (line.group === group) {
+          var expT = _.findEl(plotEl, 'template.experiment');
+          var expEl = _.cloneTemplate(expT);
 
+          expEl.setAttribute('transform', 'translate(' + plotEl.dataset.groupLineOffset + ',' + currGY + ')');
 
+          _.fillEls(expEl, '.expname', line.title);
+          if (line.or != null) {
+            groupMembers += 1;
+            _.fillEls(expEl, '.or.lcl.ucl', Math.exp(line.or).toPrecision(3) + ' [' + Math.exp(line.lcl).toPrecision(3) + ", " + Math.exp(line.ucl).toPrecision(3) + ']');
+            _.fillEls(expEl, '.wt', '' + (Math.round(line.wt / sumOfWt * 1000)/10) + '%');
+            _.fillEls(expEl, '.wtg', ', ' + (Math.round(line.wtg)/10) + '%');
+            _.setAttrs(expEl, 'line.confidenceinterval', 'x1', getX(line.lcl));
+            _.setAttrs(expEl, 'line.confidenceinterval', 'x2', getX(line.ucl));
 
-    // put summary into the plot
-    if (!isNaN(aggregates.or*0)) {
-      var sumT = _.findEl(plotEl, 'template.summary');
-      var sumEl = _.cloneTemplate(sumT);
+            _.setAttrs(expEl, 'rect.weightbox', 'x', getX(line.or));
+            _.setAttrs(expEl, 'rect.weightbox', 'width', getBoxSize(line.wt));
+            _.setAttrs(expEl, 'rect.weightbox', 'height', getBoxSize(line.wt));
 
-      sumEl.setAttribute('transform', 'translate(' + plotEl.dataset.padding + ',' + currY + ')');
+            groupAggregates.or += line.or;
+            groupAggregates.lcl += line.lcl;
+            groupAggregates.ucl += line.ucl;
+            groupAggregates.wt += (Math.round(line.wt / sumOfWt * 1000)/10);
+          } else {
+            expEl.classList.add('invalid');
+            hasInvalid = true;
+            groupHasInvalid = true;
+          }
 
-      _.fillEls(sumEl, '.or', Math.exp(aggregates.or).toPrecision(3));
-      _.fillEls(sumEl, '.lcl', "" + Math.exp(aggregates.lcl).toPrecision(3) + ",");
-      _.fillEls(sumEl, '.ucl', Math.exp(aggregates.ucl).toPrecision(3));
+          groupExperimentsEl.appendChild(expEl);
 
-      var lclX = getX(aggregates.lcl);
-      var uclX = getX(aggregates.ucl);
-      var orX = getX(aggregates.or);
-      if ((uclX - lclX) < parseInt(plotEl.dataset.minDiamondWidth)) {
-        var ratio = (uclX - lclX) / parseInt(plotEl.dataset.minDiamondWidth);
-        lclX = orX + (lclX - orX) / ratio;
-        uclX = orX + (uclX - orX) / ratio;
+          currGY += parseInt(plotEl.dataset.lineHeight);
+        }
+      });
+
+      // if stat != wt, then divide the groupAggregate sums by the number of valid lines in the group.
+      for (var stat in groupAggregates) {
+        if (stat != "wt") {
+          groupAggregates[stat] /= groupMembers;
+        }
+        else {
```

Comment:
you could probably use Number::toFixed(1) below, if your code basically does that; this kind of code (i.e. accessing some specific element in an array of something?) needs more detail in the comments, otherwise it is difficult to glance at it and understand what it is doing, and more difficult to maintain and change.